app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
+  before_action :logged_in?, only: %i[show]
```

Comment:
- [ ] **[OPTIONAL]** There is already helper method for devise that require login [`authenticate_user!`](https://github.com/plataformatec/devise#controller-filters-and-helpers)